DQMOffline/JetMET/interface/JetAnalyzer_HeavyIons_matching.h

```diff
@@ -0,0 +1,180 @@
+#include <utility>
+
+
+class MonitorElement;
+
+class JetAnalyzer_HeavyIons_matching : public DQMEDAnalyzer {
+ public:
+
+  explicit JetAnalyzer_HeavyIons_matching (const edm::ParameterSet&);
+  virtual ~JetAnalyzer_HeavyIons_matching();
+  
+  void analyze(const edm::Event&, const edm::EventSetup&) override; 
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  
+  // calojet1token = vscalo 
+  // calojet2token = pucalo 
+
+ private:
+  
+  edm::InputTag   mInputJet1Collection;
+  edm::InputTag   mInputJet2Collection;
+  
+  std::string     mOutputFile;
+  std::string     JetType1;
+  std::string     JetType2;
+  double          mRecoJetPtThreshold;
+  double          mRecoDelRMatch; 
+  double          mRecoJetEtaCut; 
+  
+  //Tokens
+  edm::EDGetTokenT<reco::CaloJetCollection> caloJet1Token_;
+  edm::EDGetTokenT<reco::CaloJetCollection> caloJet2Token_;
+  edm::EDGetTokenT<reco::PFJetCollection> pfJetsToken_;
+  edm::EDGetTokenT<reco::BasicJetCollection> basicJetsToken_;
+  edm::EDGetTokenT<reco::JPTJetCollection> jptJetsToken_;
+
+  MonitorElement * mpT_ratio_Jet1Jet2;
+  MonitorElement * mpT_Jet1_matched;
+  MonitorElement * mpT_Jet2_matched;
+  MonitorElement * mpT_Jet1_unmatched;
+  MonitorElement * mpT_Jet2_unmatched; 
+
+  MonitorElement * mHadEnergy_Jet1_unmatched;
+  MonitorElement * mEmEnergy_Jet1_unmatched;
+  MonitorElement * mChargedHadronEnergy_Jet1_unmatched;
+  MonitorElement * mNeutralHadronEnergy_Jet1_unmatched;
+  MonitorElement * mChargedEmEnergy_Jet1_unmatched;
+  MonitorElement * mNeutralEmEnergy_Jet1_unmatched;
+  MonitorElement * mChargedMuEnergy_Jet1_unmatched;
+
+  MonitorElement * mHadEnergy_Jet2_unmatched;
+  MonitorElement * mEmEnergy_Jet2_unmatched;
+  MonitorElement * mChargedHadronEnergy_Jet2_unmatched;
+  MonitorElement * mNeutralHadronEnergy_Jet2_unmatched;
+  MonitorElement * mChargedEmEnergy_Jet2_unmatched;
+  MonitorElement * mNeutralEmEnergy_Jet2_unmatched;
+  MonitorElement * mChargedMuEnergy_Jet2_unmatched;
+
+  struct MyJet{
+    int   id;
+    float pt;
+    float eta;
+    float phi;
+  };
+
+
+  typedef std::pair< MyJet, MyJet > ABJetPair;
+
+  struct CompareMatchedJets {
+    //! A-B jet match
+    bool operator()(const ABJetPair &A1, const ABJetPair &A2){
+      MyJet jet1_pair1 = A1.first; //! Jet1 1st pair
+      MyJet jet2_pair1 = A1.second; //! Jet2 1st pair
+      MyJet jet1_pair2 = A2.first; //! Jet1 2nd pair
+      MyJet jet2_pair2 = A2.second; //! Jet2 2nd pair
+      float delr1 = deltaRR(jet1_pair1.eta, jet1_pair1.phi, jet2_pair1.eta, jet2_pair1.phi);
+      float delr2 = deltaRR(jet1_pair2.eta, jet1_pair2.phi, jet2_pair2.eta, jet2_pair2.phi);
+
+      return ((delr1 < delr2) && (jet1_pair1.pt > jet1_pair2.pt));
+
+    }
+  };
+
+
+  typedef std::multiset< ABJetPair, CompareMatchedJets > ABMatchedJets;
+  typedef std::multiset< ABJetPair >::iterator ABItr;
+
+  static float deltaRR(float eta1, float phi1, float eta2, float phi2)
```

Comment:
One can't make a static method const since it isn't dependent upon the object. From a thread safety issue, static functions which do not change any global objects are perfectly safe.